testrail/api.py

```diff
@@ -638,7 +638,7 @@ def _post(self, uri, data={}):
 
     def _payload_gen(self, fields, data):
         payload = dict()
-        for field in fields:
-            if data.get(field) is not None:
-                payload[field] = data.get(field)
+        for k, v in data.items():
+            if v is not None:
+                payload[k] = v
```

Comment:
The original `_payload_gen` allowed the calling method to specify which fields from `data` should be included in the payload. Your update ignores that entirely and adds everything in `data` to the payload. I'm not sure that is desirable.